app/services/authentication/providers/forem.rb

```diff
@@ -0,0 +1,50 @@
+module Authentication
+  module Providers
+    # GitHub authentication provider, uses omniauth-github as backend
+    class Forem < Provider
+      OFFICIAL_NAME = "Forem".freeze
+      DOMAIN_URL = ApplicationConfig["PASSPORT_OAUTH_URL"] || "https://passport.forem.com".freeze
+      SETTINGS_URL = "#{DOMAIN_URL}/oauth/authorized_applications".freeze
+
+      def new_user_data
+        {
+          email: info.email,
+          name: info.name,
+          remote_profile_image_url: raw_info.remote_profile_image_url,
+          forem_username: info.user_nickname
+        }
+      end
+
+      def existing_user_data
+        {
+          email: info.email,
+          name: info.name,
+          forem_username: info.user_nickname
+        }
+      end
+
+      delegate :user_nickname, to: :info
+
+      def self.official_name
+        OFFICIAL_NAME
+      end
+
+      def self.settings_url
+        SETTINGS_URL
+      end
+
+      def self.sign_in_path(**kwargs)
+        ::Authentication::Paths.sign_in_path(
+          provider_name,
+          **kwargs,
+        )
+      end
+
+      protected
+
+      def cleanup_payload(auth_payload)
```

Comment:
Maybe we can add a comment here explaining why we need this no-op method. I assume something in our flow depends on it being defined but it's not obvious to readers of this file since the method is never called in here.